rust/src/domains/polars/frame/test.rs

```diff
@@ -39,3 +39,136 @@ fn test_margin() -> Fallible<()> {
 
     Ok(())
 }
+
+#[test]
+fn test_get_margin_max_partition_descriptors() -> Fallible<()> {
+    let lf_domain = LazyFrameDomain::new(vec![
+        SeriesDomain::new("A", AtomDomain::<i32>::default()),
+        SeriesDomain::new("B", AtomDomain::<i32>::default()),
+    ])?
+    .with_margin(
+        &["A"],
+        Margin::default()
+            .with_max_partition_length(10)
+            .with_max_partition_contributions(3),
+    )?;
+
+    assert_eq!(
+        lf_domain
+            .get_margin(BTreeSet::from(["A".to_string(), "B".to_string()]))
+            .max_partition_length,
+        Some(10)
+    );
+
+    assert_eq!(
+        lf_domain
+            .get_margin(BTreeSet::from(["B".to_string()]))
+            .max_partition_length,
+        None
+    );
+
+    assert_eq!(
+        lf_domain
+            .get_margin(BTreeSet::from(["A".to_string(), "B".to_string()]))
+            .max_partition_contributions,
+        Some(3)
+    );
+
+    assert_eq!(
+        lf_domain
+            .get_margin(BTreeSet::from(["B".to_string()]))
+            .max_partition_contributions,
+        None
+    );
+    Ok(())
+}
+
+#[test]
+fn test_get_margin_covering_small_to_large() -> Fallible<()> {
+    let lf_domain = LazyFrameDomain::new(vec![
+        SeriesDomain::new("A", AtomDomain::<i32>::default()),
+        SeriesDomain::new("B", AtomDomain::<i32>::default()),
+    ])?
+    .with_margin(
+        &["A"],
+        Margin::default()
+            .with_max_num_partitions(10)
+            .with_max_influenced_partitions(3),
+    )?
+    .with_margin(
+        &["B"],
+        Margin::default()
+            .with_max_num_partitions(10)
+            .with_max_influenced_partitions(3),
+    )?;
+
+    assert_eq!(
+        lf_domain
+            .get_margin(BTreeSet::from(["A".to_string(), "B".to_string()]))
+            .max_num_partitions,
+        Some(100)
+    );
+
+    assert_eq!(
+        lf_domain
+            .get_margin(BTreeSet::from(["B".to_string()]))
+            .max_influenced_partitions,
+        Some(3)
+    );
```

Comment:
Just to check the extreme cases, would it be good to check what the behavior is on an empty set? or if a member of the set is not in the margins?